site/content/how-to/traffic-management/snippets.md

```diff
@@ -0,0 +1,473 @@
+
+- Save the public IP address and port of NGINX Gateway Fabric into shell variables:
+
+   ```text
+  GW_IP=<ip address>
+  GW_PORT=<port number>
+  ```
+
+  {{< note >}} In a production environment, you should have a DNS record for the external IP address that is exposed, and it should refer to the hostname that the gateway will forward for. {{< /note >}}
+
+- Create the coffee and tea example applications:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/app.yaml
+  ```
+
+- Create a Gateway:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/gateway.yaml
+   ```
+
+- Create HTTPRoutes for the coffee and tea applications:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/httproutes.yaml
+   ```
+
+- Test the configuration:
+
+  You can send traffic to the coffee and tea applications using the external IP address and port for NGINX Gateway Fabric.
+
+  Send a request to coffee:
+
+  ```shell
+  curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee
+  ```
+
+  This request should receive a response from the coffee Pod:
+
+  ```text
+  Server address: 10.244.0.7:8080
+  Server name: coffee-76c7c85bbd-cf8nz
+  ```
+
+  Send a request to tea:
+
+  ```shell
+  curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/tea
+   ```
+
+  This request should receive a response from the tea Pod:
+
+  ```text
+  Server address: 10.244.0.6:8080
+  Server name: tea-76c7c85bbd-cf8nz
+  ```
+
+  Before we enable rate limiting, try sending multiple requests to coffee:
+
+  ```shell
+  for i in `seq 1 10`; do curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee; done
+  ```
+
+  You should see all successful responses in quick succession as we configured any rate limiting rules yet.
+
+---
+
+## Create Rate Limiting SnippetsFilters
+
+Configure a rate limiting `SnippetsFilter` named `rate-limiting-sf` by adding the following `SnippetsFilter`:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.nginx.org/v1alpha1
+kind: SnippetsFilter
+metadata:
+  name: rate-limiting-sf
+spec:
+  snippets:
+    - context: http
+      value: limit_req_zone \$binary_remote_addr zone=rate-limiting-sf:10m rate=1r/s;
+    - context: http.server.location
+      value: limit_req zone=rate-limiting-sf burst=3;
+EOF
+```
+
+This `SnippetsFilter` defines two Snippets to configure rate limiting. The first Snippet injects the value: `limit_req_zone \$binary_remote_addr zone=rate-limiting-sf:10m rate=1r/s;`
```

Comment:
Do you think "will inject" instead of "injects" makes more sense? I'm on the fence. It does fit with the sentence on line 152 "This SnippetsFilter will limit..."

Reply:
ah gosh I'm on the fence too, but adding in an additional two "will" seems annoying to read hahaha